src/main.cpp

```diff
@@ -2696,6 +2811,18 @@ bool DisconnectBlock(CBlock& block, CValidationState& state, CBlockIndex* pindex
                 // resurrect prevBlockTopQualityCertHash bwts
                 assert(blockUndo.scUndoDatabyScId.at(cert.GetScId()).contentBitMask & CSidechainUndoData::AvailableSections::SUPERSEDED_CERT_DATA);
                 view.RestoreBackwardTransfers(prevBlockTopQualityCertHash, blockUndo.scUndoDatabyScId.at(cert.GetScId()).lowQualityBwts);
+
+#ifdef ENABLE_ADDRESS_INDEXING
+                // Set the lower quality BTs as top quality
+                if (fAddressIndex && indexesProcessing == flagBlockProcessingType::COMPLETE)
+                {
+                    CTxIndexValue txIndexVal;
+                    assert(pblocktree->ReadTxIndex(prevBlockTopQualityCertHash, txIndexVal));
+
+                    view.UpdateBackwardTransferIndexes(prevBlockTopQualityCertHash, txIndexVal.txIndex, addressIndex, addressUnspentIndex,
+                                                       CCoinsViewCache::flagIndexesUpdateType::RESTORE_CERTIFICATE);
+                }
+#endif // ENABLE_ADDRESS_INDEXING
             }
```

Comment:
Update also txIndex marking prevBlockTopQualityCertHash the certMaturityHeight positive